### PR TITLE
Produce textual feedback on transfer process

### PR DESCRIPTION
**Pull request by martindurant**
(controlled using verbose= flag in constructor)

### CODE REVIEW

File: adlfs/multithread.py [hunk 1]
```diff
@@ -69,9 +69,10 @@ class ADLDownloader(object):
     adlfs.transfer.ADLTransferClient
     """
     def __init__(self, adlfs, rpath, lpath, nthreads=None, chunksize=2**28,
-                 blocksize=2**22, client=None, run=True, overwrite=False):
+                 blocksize=2**22, client=None, run=True, overwrite=False,
+                 verbose=True):
         if not overwrite and adlfs.exists(rpath):
```
Comment (begoldsm) [inline adlfs/multithread.py]: I am confused by this logic. Shouldn't it be checking for the existence of the local file? Why are the checks inverted for upload and download based on the errors being raised.

Comment (martindurant) [inline adlfs/multithread.py]: We had it the wrong way around - switched in this PR

**Comment by begoldsm** [comment]
@martindurant is there a test for overwrite= false that validates the right path is being returned in the `FileExistsError`? That should be added here as a regression test since that was the bug previously. Also that it properly fails when the right file already exists, since before it complained when the source file existed.

**Comment by martindurant** [comment]
See tests/test_multithread.py , test_upload_overwrite, test_download_overwrite

**Comment by begoldsm** [comment]
Great, thanks @martindurant! I will hold off on the merge until that test is added to this PR.

**Pull request merged**